src/pages/ScholarshipForm.jsx

```diff
@@ -1,115 +1,134 @@
-  const [values, setValues] = useState({
-    name: '',
-    deadline: null,
-    description: '',
-    amountType: null,
-    minAmount: 0,
-    maxAmount: 0,
-    website: '',
-    scholarshipType: '',
-  });
-
-  const [errors, setErrors] = useState({});
 
-  function validateForm() {
-    /* eslint-disable no-restricted-globals */
-    setErrors({
-      name: !values.name,
-      deadline: !values.deadline || isNaN(values.deadline),
-      description: !values.description,
-      website: !websiteReg.test(values.website),
-      amountHelperText: invalidAmountFields(
-        values.amountType,
-        values.minAmount,
-        values.maxAmount
-      ),
-    });
-  }
+  const formik = useFormik({
+    initialValues: {
+      name: '',
+      deadline: null,
+      description: '',
+      amount: {
+        type: null,
+        minAmount: 0,
+        maxAmount: 0,
+      },
+      website: '',
+    },
+    validationSchema,
+    onSubmit: (values, { setSubmitting, resetForm }) => {
+      setSubmitting(true);
+      alert(JSON.stringify(values, null, 2));
+      setSubmitting(false);
+      resetForm();
+    },
+  });
 
   function updateAmount(minAmount, maxAmount) {
-    setValues({ ...values, minAmount, maxAmount });
+    formik.setFieldValue('amount.minAmount', minAmount, false);
+    formik.setFieldValue('amount.maxAmount', maxAmount, false);
   }
 
   return (
     <Container maxWidth="md">
-      <form className={classes.containerStyle}>
+      <form className={classes.containerStyle} onSubmit={formik.handleSubmit}>
         <h1>Submit a Scholarship</h1>
         <div>
           <TextField
+            className={classes.fieldStye}
             id="name"
-            label="Scholarship Name"
-            error={errors.name}
-            helperText={errors.name ? 'Please enter a name' : ' '}
-            required
-            value={values.name}
-            onChange={(e) => setValues({ ...values, name: e.target.value })}
+            label="Scholarship Name *"
+            error={formik.touched.name && Boolean(formik.errors.name)}
+            helperText={(formik.touched.name && formik.errors.name) || ' '}
+            value={formik.values.name}
+            onChange={formik.handleChange}
             fullWidth
           />
         </div>
+
         <DatePicker
+          className={classes.fieldStye}
           id="deadline"
-          label="Deadline"
-          error={errors.deadline}
-          helperText={(errors.deadline && 'Please enter a valid date') || ' '}
-          value={values.deadline}
-          onChange={(date) => setValues({ ...values, deadline: date })}
+          label="Deadline *"
+          error={formik.touched.deadline && Boolean(formik.errors.deadline)}
+          helperText={
+            (formik.touched.deadline && formik.errors.deadline) || ' '
+          }
+          value={formik.values.deadline}
+          onChange={(date) => formik.setFieldValue('deadline', date, false)}
```

Comment:
The third parameter is to validate on change, why set it to false?

Reply:
I had thought It would be better to validate when the user tried submitting the form again but since the other fields validate `onChange`, setting the validation to true makes more sense.